### PR TITLE
Random erasing applied to mask keys

### PR DESCRIPTION
**Pull request by miquelmarti**
#### Changes

Applies random erasing to mask keys as well for dense prediction problems.

For simplicity for now, always erase masks using zeros. This should be an option for each key instead, since ignore or invalid values for different keys can be different and should be used for padding values in affine transformations too.

Also, adds test case to cover random erasing with mask keys and fixes typo in file name ('earasing -> erasing').

TODO: Apply to keypoints and bounding boxes.
This raises a more general question of what to do with bounding boxes and points in invalid regions of the image. Also pointed out in https://github.com/kornia/kornia/issues/1522: 

> while it may make sense to return bounding boxes with negative values to support inverting a transformation, bbox_xyxy or bbox_xywh require valid image coordinates, so a clipping utility or a parameter that enables this would be cool

In my opinion, something similar to this should be done for random erasing too.

#### Type of change

- [ ] 📚  Documentation Update
- [x] 🧪 Tests Cases
- [ ] 🐞 Bug fix (non-breaking change which fixes an issue)
- [x] 🔬 New feature (non-breaking change which adds functionality)
- [ ] 🚨 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 📝 This change requires a documentation update

#### Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] Did you update CHANGELOG in case of a major change?

### CODE REVIEW

**Review by shijianjian — APPROVED** [review]
Acceptable for now. I would try avoid increasing the complexity of those utility files. Given the fact that we will have many different augmentations that may have different solutions to different data keys. Any suggestions?

File: test/augmentation/test_container.py [hunk 1]
```diff
@@ -350,6 +350,18 @@ def test_random_crops_and_flips(self, device, dtype):
...
+        fill_value = 0.5
+        input = torch.randn(3, 3, 100, 100, device=device, dtype=dtype)
+        aug = K.AugmentationSequential(
+            K.RandomErasing(p=1., value=fill_value), data_keys=["input", "mask"],
+        )
+
+        reproducibility_test((input, input), aug)
+
+        out = aug(input, input)
+        assert torch.all(out[1][out[0] == fill_value] == 0.)
```
Comment (edgarriba) [inline test/augmentation/test_container.py:363]: Safer to use `asset_close`

**Review by edgarriba — COMMENTED** [review]
Seems something went wrong with another test. I’d suggest to run locally `pytest test/augmentation`a

**Comment by miquelmarti** [comment]
> Acceptable for now. I would try avoid increasing the complexity of those utility files. Given the fact that we will have many different augmentations that may have different solutions to different data keys. Any suggestions?

Honestly, I don't really see a cleaner way around it...

I'll fix the failing tests.

**Pull request merged**